ops/charm.py

```diff
@@ -995,6 +1081,23 @@ def config(self) -> model.ConfigData:
         return self.model.config
 
 
+def _evaluate_status(charm: CharmBase):  # pyright: ignore[reportUnusedFunction]
+    """Trigger collect-status events and evaluate and set the highest-priority status.
+
+    See :class:`CollectStatusEvent` for details.
+    """
+    if charm.framework.model._backend.is_leader():
+        charm.on.collect_app_status.emit()
+        app = charm.app
+        if app._collected_statuses:
+            app.status = model.StatusBase._get_highest_priority(app._collected_statuses)
```

Comment:
Could leadership status (on the juju side) change mid-hook?
A contrived example is:
- `is_leader()` is True, because that's how we started
- charm code does `time.sleep(LEASE_RENEWAL_PERIOD + 1)`

If that's a possible failure, I'm not sure putting the charm in error state is the best course of action.

Reply:
Yeah, I suppose that's possible, but should be very unlikely. And if it does happen, the update-status or another hook on the new leader will bring it back into line. I don't think we need to do anything here for now.